scipy/stats/_multivariate.py

```diff
@@ -5685,3 +5687,625 @@ def _sample_uniform_direction(dim, size, random_state):
     samples = random_state.standard_normal(samples_shape)
     samples /= np.linalg.norm(samples, axis=-1, keepdims=True)
     return samples
+
+
+_vonmises_fisher_doc_default_callparams = """\
+mu : array_like
+    Mean direction of the distribution. Must be a one-dimensional unit
+    vector of norm 1.
+kappa : float
+    Concentration parameter. Must be positive."""
+
+_vonmises_fisher_doc_callparams_note = "bla"
+
+_vonmises_fisher_doc_frozen_callparams = "test"
+
+_vonmises_fisher_doc_frozen_callparams_note = """\
+See class definition for a detailed description of parameters."""
+
+vonmises_fisher_docdict_params = {
+    '_doc_vonmises_fisher_default_callparams':
+        _vonmises_fisher_doc_default_callparams,
+    '_doc_vonmises_fisher_callparams_note':
+        _vonmises_fisher_doc_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+vonmises_fisher_docdict_noparams = {
+    '_doc_vonmises_fisher_default_callparams':
+        _vonmises_fisher_doc_frozen_callparams,
+    '_doc_vonmises_fisher_callparams_note':
+        _vonmises_fisher_doc_frozen_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+
+class vonmises_fisher_gen(multi_rv_generic):
+    r"""A von Mises-Fisher variable.
+
+    The `mu` keyword specifies the mean direction vector. The `kappa` keyword
+    specifies the concentration parameter.
+
+    Methods
+    -------
+    pdf(x, mu=None, kappa=1)
+        Probability density function.
+    logpdf(x, mu=None, kappa=1)
+        Log of the probability density function.
+    rvs(mu=None, kappa=1, size=1, random_state=None)
+        Draw random samples from a von Mises-Fisher distribution.
+    entropy()
+        Compute the differential entropy of the von Mises-Fisher distribution.
+    fit(data)
+        Fit a von Mises-Fisher distribution to data.
+
+    Parameters
+    ----------
+    %(_doc_random_state)s
+
+    See Also
+    --------
+    scipy.stats.vonmises : Von-Mises Fisher distribution in 2D on a circle
+
+    Notes
+    -----
+    The von Mises-Fisher distribution is a directional distribution on the
+    n-dimensional unit sphere. The probability density function of a
+    unit vector :math:`\mathbf{x}` is
```

Comment:
I think it's still missing the word `surface`. From the text we get that `mu` is a point on the surface because we have 2 info: `on` and `unit vector`. But I think we cannot rely on using the word `on` as not all readers speak good English. We need to be clear that there are no points inside the hypersphere.